added a syncEvents method

ccorman opened the pull request:
Like the title said, i added a syncEvents method, because i needed it, and it was also asked as enhancements. It is my first pull request, so i hope everything is okay :)

ccorman committed:
added a syncEvents method

jasonjoh commented on calendar-api.js:
```diff
@@ -91,6 +91,119 @@ module.exports = {
+   *   endDateTime: endDateTime
+   * };
+   *
+   * outlook.mail.syncEvents(apiOptions, function(error, events) {
```
This should be `outlook.calendar.syncEvents`, right? :)

ccorman replied:
Yup, you're right, I copied / paste doc from mail-api and I forgot to change this ^^'

jasonjoh commented on calendar-api.js:
```diff
@@ -91,6 +91,119 @@ module.exports = {
+   * 
+   * var apiOptions = {
+   *   token: token,
+   *   calendarId: 'Primary', // Not mandatory, the Primary calendar will be used by default, but you can specify another calendar Id
```
I'm not sure that `Primary` is a valid value here.

ccorman replied:
You're right again, I was thinking about Google calendar API, but it does not seem to exist with Outlook API.

jasonjoh commented on calendar-api.js:
```diff
@@ -91,6 +91,119 @@ module.exports = {
+  
+  syncEvents: function(parameters, callback) {
+      var userSpec = utilities.getUserSegment(parameters);
+      var calendarSpec = parameters.calendarId === undefined ? '' : "/calendars('" + parameters.calendarId + "')"
```
This doesn't seem to be used anywhere? It should come in `requestUrl` right after `userSpec`.

ccorman replied:
That's it, and also, it shouldn't be /calendars(calendar_id), but /calendars/calendar_id, like this:
`var calendarSpec = parameters.calendarId === undefined ? '' : "/calendars/" + parameters.calendarId;`

jasonjoh commented:
Thanks @ccorman! It looks pretty good, but I left a few notes for some minor issues. Can you update?

ccorman committed:
Update after notes on minor issues

- 'primary' value is not valid
- this is of course outlook.calendar, and not outlook.mail
- calendarSpec should be /calendars/calendar_id instead of /calendars(calendar_id), and i added it to the request url

ccorman commented:
Thanks for the notes, I updated the code :)

jasonjoh commented:
Thanks! I've gone ahead and merged this into the source, but it will be a little while before I can get a build together and push to NPM. Too many competing priorities for upcoming //BUILD conference :).

I'll leave #25 open for now until I can push to NPM.
